### PR TITLE
Upgrade to vm2

### LINKED ISSUE #314: [core][cli] Upgrade to vm2
Currently, we are using the node `vm` module to load and execute WoT applications. However, `vm` has different problems:
1. it is not completely "safe". [Here](https://github.com/patriksimek/vm2#vm2-----) is explained how a malevolent script can access the main context and kill the servient. 
2. it is difficult to make `require` works. As discussed in different issues/PRs is not possible to require local modules or scripts without heavy modification to the current code. Modifications that will end up very similar to what is already implemented in vm2.

So my suggestion is to upgrade to [vm2](https://github.com/patriksimek/vm2) which has interesting features  (from the readme):
* Runs untrusted code securely in a single process with your code side by side
* Full control over sandbox's console output
* Sandbox has limited access to process's methods
* Sandbox can require modules (builtin and external)
* You can limit access to certain (or all) built-in modules (*currently is not working in our servient CLI*)
* You can securely call methods and exchange data and callbacks between sandboxes
* Is immune to all known methods of attacks
* Transpilers support (*so we may support TS from the cli without the building step*)

What do you think? I can volunteer to make this happen.

### PR DESCRIPTION
**Pull request by relu91**
This PR closes #314. As described in the issue vm2 has a couple of benefits. In particular, it will solve problems for `require` calls in WoT application scripts ( #191). With subsequent PRs we could close issues #190 #192 #275, without major modifications. Probably also #185 might be resolved thanks to the `transpilers` support of vm2.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -24,12 +24,13 @@
...
-    "wot-typescript-definitions": "0.7.3",
     "@node-wot/td-tools": "0.7.2",
-    "uuid": "3.4.0",
+    "@types/uritemplate": "0.3.4",
     "rxjs": "5.5.11",
     "uritemplate": "0.3.4",
-    "@types/uritemplate": "0.3.4"
+    "uuid": "3.4.0",
+    "vm2": "^3.9.2",
+    "wot-typescript-definitions": "0.7.3"
```
Comment (danielpeintner) [inline packages/core/package.json:33]: May I ask you to introduce back the original order since most of the entries are just re-shuffled and not *new* expect "vm2": "^3.9.2"

The reason I am asking for it is that if we merge THIS work into https://github.com/eclipse/thingweb.node-wot/tree/v0.8.x we will get merge problems since this will be detected as a CHANGE.

File: packages/browser-bundle/package.json [hunk 2]
```diff
@@ -22,6 +22,6 @@
   "dependencies": {
   },
   "scripts": {
-    "build": "browserify index.js --plugin tinyify -o dist/wot-bundle.min.js"
+    "build": "browserify index.js --plugin tinyify --external coffee-script -o dist/wot-bundle.min.js"
```
Comment (danielpeintner) [inline packages/browser-bundle/package.json:25]: Is this change needed?

**Review by danielpeintner — CHANGES_REQUESTED** [review]
Did some sanity checks/runs and everything seems to work just fine. Great work, thanks!

I added 2 comments/requests....

File: packages/core/package.json (hunk 1 above)
Comment (relu91) [reply packages/core/package.json:33]: yes sure I'll try to put everything back in order. It was npm that changed the order.

File: packages/browser-bundle/package.json (hunk 2 above)
Comment (relu91) [reply packages/browser-bundle/package.json:25]: Yes, vm2 has as *optional* dependency coffee-script. However, broswerify does not detect it as optional and tries to find the module. The --external argument just tells to broswerify to ignore that module since it will be added at runtime by users of the library (although I do not think this will ever happen in our case).

Comment (danielpeintner) [reply packages/browser-bundle/package.json:25]: I see. Let's keep `--external coffee-script`. I was just not sure *why* it is there

File: packages/core/package.json (hunk 1 above)
Comment (relu91) [reply packages/core/package.json]: It should be fine now 👍

**Review by danielpeintner — APPROVED** [review]
